### PR TITLE
Update VSCode to 1.85.2, including updates to TypeScript

### PR DESCRIPTION
**Pull request by penalosa**
**What this PR solves / how to test:**

This updates the set of VSCode patches for the quick editor to be based of 1.85.2, rather than 1.76.0.

This can be tested by modifying the Quick Edit URL in a local copy of the Workers Playground or the dashboard to point to the prerelease build generated by this PR (https://cdf6467f.quick-edit-cny.pages.dev/). Particular attention should be paid to TypeScript (through JSDoc) support.

**Author has addressed the following:**

- Tests
  - [ ] Included
  - [x] Not included because: Must be manually tested
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] Included
  - [x] Not necessary because: Not relevant
- Associated docs
  - [ ] Issue(s)/PR(s):
  - [x] Not necessary because: Not relevant

### CODE REVIEW

File: packages/quick-edit/package.json [hunk 1]
```diff
@@ -25,6 +25,6 @@
 		"wrangler": "workspace:*"
 	},
 	"engines": {
-		"node": "<17"
+		"node": ">=18.15"
```
Comment (CarmenPopoviciu) [inline packages/quick-edit/package.json:28]: not `18.18`? What does this version number depend on?

Comment (penalosa) [reply packages/quick-edit/package.json:28]: Thanks for catching this! This should probably be 18.18 too

File: packages/quick-edit/functions/_middleware.ts [hunk 2]
```diff
@@ -5,6 +5,8 @@ export const onRequest = async ({
 }: Parameters<PagesFunction>[0]) => {
 	const url = new URL(request.url);
 
+	const isLocalDev = url.hostname === "localhost";
```
Comment (petebacondarwin) [inline packages/quick-edit/functions/_middleware.ts:8]: Is this a reliable enough way to compute local dev mode?
What if it was 127.0.0.1 or similar?
Do we care? Perhaps not.

File: packages/quick-edit/package.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
 	"author": "workers-devprod@cloudflare.com",
 	"scripts": {
 		"custom:build": "./build.sh",
-		"dev": "concurrently '../../vendor/vscode/scripts/code-web.sh --extensionDevelopmentPath ../../packages/quick-edit/web/quick-edit-extension' 'npm --prefix web/quick-edit-extension run watch-web' 'yarn --cwd ../../vendor/vscode watch' 'yarn --cwd ../../vendor/vscode watch-web'",
+		"dev": "concurrently 'pnpm exec wrangler pages dev ./web' 'npm --prefix web/quick-edit-extension run watch-web' 'yarn --cwd ../../vendor/vscode watch' 'yarn --cwd ../../vendor/vscode watch-web'",
```
Comment (petebacondarwin) [inline packages/quick-edit/package.json:14]: So we have swapped out a built in web server for a local pages server?
Anything more to it than that?

**Review by petebacondarwin — APPROVED** [review]
I have no idea how to go about reviewing all those patch changes!
But otherwise this seems like a good change.
How do we test this stuff?

**Comment by petebacondarwin** [comment]
Without a changeset we will no longer deploy this to production, I believe.

File: packages/quick-edit/functions/_middleware.ts (hunk 2 above)
Comment (penalosa) [reply packages/quick-edit/functions/_middleware.ts:8]: We don't really care—this is just for local development and it'll probably always be `localhost`

File: packages/quick-edit/package.json (hunk 3 above)
Comment (penalosa) [reply packages/quick-edit/package.json:14]: Yep, that's all that changed here

**Comment by penalosa** [comment]
> I have no idea how to go about reviewing all those patch changes! But otherwise this seems like a good change. How do we test this stuff?

The best way to test this is to run a local copy of the workers playground, and modify the editor URL to point to the prerelease from this branch, and then play around and try and break it

**Comment by lrapoport-cf** [comment]
@petebacondarwin will you be able to test this per @penalosa 's instructions?

**Comment by petebacondarwin** [comment]
I tried this out locally. It seems to be working although I couldn't identify any of the new VS code features in our usage. For example, interactive inlay hints or new refactorings. But these might just not be available in our current setup.

**Comment by petebacondarwin** [comment]
Needs a rebase then it is good to land.

**Pull request merged**